### DISCUSSION TITLE
진통제 같은 서비스 만들기

### DISCUSSION THREAD

**Discussion by Choi-JJunho** [woowacourse-teams/2023-pium#105] [뜨거운 토론]
# 미니멈, 타겟, 에픽

> 미니멈
프로젝트가 성공하기 위해 반드시 일어나야 하는 최소한의 일. 성실하게 작업을 수행한다면 거의 90% 확실하게 일어날 수 있는 일들. 우리가 반드시 이 고통은 해결해야 한다.

미니멈: 리마인더와 타임라인을 이용한 반려 식물 관리 이력을 기록하고 확인하는 기능을 제공한다. 3차 데모데이 때 구현되어있어야하며 개선되어야한다.

> 타겟
이런 일이 일어나기를 바라는 것. 어렵지만 달성 가능한 목표여야 함. 부지런히 노력한다면 60%이상 실현 가능성이 있어야 함

타겟: 웹 환경에서 제공하기 어려운 알림 기능을 위해 구글 캘린더와 연동하여 알림을 준다. 공공데이터에 없는 데이터를 제공하여 추가적인 사전 식물 정보를 제공한다.

> 에픽
상상을 뛰어넘는 성공의 모습입니다. 일반적으로 모든 사람이 처음에 스스로 검열하므로 이 목표를 두 번 채워 보세요. 부끄러워하지 마세요! 그리고 이 과정에서 자신이 정말 중요하게 생각하는 것을 상상해 보는 것도 좋습니다.

에픽: 리엑트 네이티브 혹은 PWA를 통해 어플로 개선한다. 이를 통해 모바일환경의 푸시알림을 구현할 수 있다.
사용자 환경을 기반으로한 식물 추천 서비스.
이미 기록이 있으면 처음 기르는 사람이 가이드로 삼을 수 있다.

cc. @yeonkkk @bassyu @Kim0914 @WaiNaat @rawfishthelgh @hozzijeong

**Comment by Choi-JJunho** [comment]
## 작은 단위 스터디 만드는 팀 (예시)

(문단 1 : 페르소나 혹은 사용자의 문제상황을 언급하면서 서비스의 필요성을 강조하는 글을 작성한다.)

누구나 공부하다가 어려운 개념 하나에 막혀, 시간을 끌며 책을 찾아보거나 블로그를 뒤져본 적 있으실 겁니다. 이런 과정이 힘들고 진이 빠져서 결국 학습 계획이 미뤄지는 일이 흔합니다. 또 여럿이서 진행하는 스터디에서도 주제 선정, 진행 방식, 참여자 간의 진도 차이 등으로 인해 효율성이 떨어지는 경우가 많습니다. 저 또한 이런 경험들이 반복되면서 공부에 대한 흥미를 잃어버리고 한동안 슬럼프에 빠졌던 적이 있습니다. 그러다 만족스러운 학습 방법과 스터디 진행 방식을 고민하게 되었고, 효과적이었던 과정을 서비스로 녹여내어 ‘공부해’라는 서비스를 만들었습니다.

(문단 2 : 서비스의 간략한 기능을 소개한다.)
‘공부해’는 학습 과정을 게임처럼 만들어 학습 목표 달성을 도와주는 서비스입니다. 스터디를 예로 들면, 단순히 주제에 대해 이야기 나누는 것이 아니라, 스터디 플랫폼에서 제공하는 가이드라인에 따라 목표를 설정하고 이를 달성하는 것이 핵심입니다.

(문단 3 : 서비스를 사용하는 사용자의 모습을 예시로 들면서 구체적인 기능을 설명한다.)
예를 들어, 자바스크립트의 'this'를 배우려고 한다고 상상해보세요. 그럼 '공부해'가 여러분에게 'this'에 대해 어떻게 생각하는지, 간단하게 정의할 수 있는지, 또는 간단한 예제 코드에서 'this'를 어떻게 활용하는지 질문하며 학습을 시작하게 도와줍니다. 이런 질문들은 학습의 첫걸음을 떼는 데 있어 어떤 방향으로 나아갈 것인지 구체적인 가이드라인을 제시합니다.

그리고 여러분이 학습을 진행하면서, 'this'를 본인의 코드에 어떻게 적용해보았는지, 그리고 'this'를 사용함으로써 코드에서 어떤 차이점을 발견할 수 있었는지 등의 질문들로 여러분의 학습 진행 상황을 체크합니다.

마지막으로는, 'this'에 대한 이해도를 실질적으로 평가하는 퀴즈를 통해 여러분이 얼마나 학습 목표를 달성하였는지 확인할 수 있습니다.

(문단 4 : 서비스가 제공하는 가치를 다시 확인하며 차별성을 강조한다.)
이렇게 '공부해'는 단순히 정보를 제공하는 것이 아니라, 여러분의 학습 과정을 이끌어 나가는데 있어 명확하고 구체적인 가이드라인을 제공합니다. 이러한 점이 바로 '공부해'가 독특한 이유입니다.

그래서 '공부해' 서비스에서는 스터디를 만드는 그 순간부터 이미 학습이 시작됩니다. 왜냐하면 단순히 'this 스터디 모집합니다' 같은 방식으로 스터디를 만들 수 없기 때문입니다. 스터디를 만들때부터 ‘공부해’ 서비스가 제안하는 가이드라인에 따라 달성 가능한 목표를 설정해야만 합니다. 이렇게 하면 스터디 개최자는 달성 가능한 명확한 목표를 생각하며 스터디를 열고 참여자들과 소통할 수 있는 준비가 됩니다.

(문단 5 : 부가적으로 제공하는 기능들에 대해서 이야기한다.)
또한, 제한된 시간 동안 집중하여 목표를 달성하도록 도와주는 타이머 기능도 갖추고 있습니다 타이머는 단순히 시간을 보여주는 것이 아니라, 중간에 진행 상황을 묻는 질문을 던져 스터디가 잘 진행되고 있는지 체크하는 역할도 합니다.

'공부해'는 이런 과정을 통해 게임처럼 목표를 설정하고, 제한된 시간 동안 목표를 달성하는 경험을 제공하며 성장을 도와줍니다.

(문단 6 : 마무리문단)
지금 바로 '공부해' 서비스에 참여해보세요! 이 서비스는 단순한 학습 도구를 넘어, 새로운 관점과 지식, 그리고 성장의 기회를 제공합니다. 이 기회를 놓치지 마세요. 함께 '공부해'에서 성장해봅시다!

**Comment by rawfishthelgh** [comment]
# 피움 소개글
***
여러분, 혹시 식물을 길러본 경험이 있으신가요? 
설레는 마음으로 조그마한 식물을 집안에 들인 여러분은, 분명 인터넷에 검색해 봤을 거예요.

“산세베리아 물 주는 법…”
“라벤더 조도 조절…”

그러나, 분명 인터넷 식집사들의 수많은 조언을 성실하게 따라 했는데도 불구하고, 어느 순간 팍 죽어버리는 식물을 보곤 했을 겁니다.
여기서 질문, 여러분의 성실한 노력에도 불구하고, 식물은 왜 죽었을까요?

정답은 여러분의 '환경'에 존재합니다. 같은 식물이라도, 어느 환경에서 기르냐에 따라서 알맞은 관리법은 천지 차이입니다. 그래서 개개인의 환경에 알맞은 관리법은 경험을 통해서 터득할 수밖에 없어요. 

그렇다면 어떻게 그 경험을 관리법으로 발전시킬 수 있을까요? 저희가 내놓은 해답은 '기록'에 있었습니다. 여러분의 관리 경험을 쉽게 기록하는 서비스를 통해서, 각자에게 알맞은 관리법을 더 빠르게 도출해 내는 거죠. 

'피움'은 이러한 가치를 통해 탄생하게 되었습니다. 리마인더를 이용해서 반려 식물 관리 이력을 기록하고, 타임라인을 이용해서 여러분의 관리 이력을 일목요연하게 확인하는 것. 이 두 가지가 피움이 여러분께 제공하는 기능입니다.

조그마한 라벤더를 집안에 들인 당신은, 피움이 제공하는 반려 식물에 대한 정보를 참고하여 물 주기 주기를 3일로 지정합니다. 예정일이 되면 리마인더에서 물을 줘야 한다는 신호를 받게 되죠. 하지만 당신은 3일 동안 화분의 흙이 충분히 마르지 않는 느낌을 받습니다. 또 줄기가 노랗게 물든 것을 확인하게 되죠. 여러분은 '미루기' 버튼을 통해 라벤더의 물 주는 날짜를 하루 미루고, 4일마다 물을 주는 것으로 주기를 바꿨습니다. 이후 푸릇푸릇하게 자라는 라벤더를 보며 여러분만의 라벤더 관리 방법을 터득하는 거죠. 또 여러분이 어떤 과정을 통해 식물을 병들게 했는지 '관리 히스토리'가 남겨짐으로써, 이전에 했던 것과는 다른 방법을 시도할 수가 있습니다.

여러분이 피움을 사용하는 순간, 여러분이 식물을 죽인 경험마저도 하나의 가치로써 탄생하게 됩니다. 내가 어떤 과정으로 식물을 죽여왔는지 다시 파악하고, 같은 실수를 되풀이하지 않을 수 있는 거죠. 또 단순히 리마인더를 보는 것만으로 내가 어떤 식물을 관리 중인지 파악하고, 누구에게 물을 줘야 하는지, 물을 주는 것을 까먹지 않았는지, 물을 주는 것을 언제까지 미룰 것인지 버튼 하나로 기록을 할 수 있습니다.

피움의 식물 관리 정보는 공공 데이터를 기반으로 한 신뢰도 있는 정보를 제공합니다. 이를 통해 여러분은 유의미한 정보를 참고할 수 있어요.  또 히스토리가 하나의 반려 식물에 대한 관리 이력을 제공했다면, 캘린더를 활용해 여러분의 전체적인 관리 이력을 더 간단하게 확인할 수도 있습니다. 이렇게 피움은 다양한 차원에서 여러분의 기록을 확인 및 비교할 수 있어요.

 더 이상 같은 실수를 되풀이하지 마세요. 여러분이 식물을 죽여왔던 것은, 단순히 뭘 잘못했는지 '기억'하지 못했기 때문입니다. 
이제 여러분의 반려 식물에 예쁜 꽃을 피워 봅시다.

'피움'을 통해서요!

**Reply by WaiNaat** [reply]
글 재밌게 읽었습니다 👍 
스투키 <-- 이친구는 물을 자주 줘야 하는 식물이 아니라서 다른 식물로 바꾸면 좋을 것 같아요

**Comment by Kim0914** [comment]
오.. 작가이신데요?
좋습니다 😎

**Comment by bassyu** [comment]
글 좋네요! 감동했습니다 😭

**Comment by Choi-JJunho** [comment]
글 진짜 잘쓰시네요 👍
💯점 드리고 갑니다

**Comment by yeonkkk** [comment]
하마드 짱👍🌱

**Comment by hozzijeong** [comment]
좋아요!

**Comment by Choi-JJunho** [comment]
> 해당 PR로 완료된 논의사항입니다
- https://github.com/woowacourse/woowa-writing-5/pull/373
